src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,698 @@
+            revert TooSoon(block.timestamp, lastClaimTimestampCached, claimFrequencyCached);
+        }
+
+        _processRewards();
+    }
+
+    function _processRewards() internal returns (TokenAmount[] memory processed_) {
+        // 1. Claim CVX, CRV & FXS rewards
+        (uint256 totalCrv, uint256 totalCvx, uint256 totalFxs) = claimAndRegisterRewards();
+
+        // 2. Swap all rewards for DAI
+        // NOTE: assume that always will be crv & cvx to convert to stables, while fxs depends on private vaults
+        uint256 totalDaiEarned;
+
+        swapCrvForWeth(totalCrv);
+        swapCvxForWeth(totalCvx);
+        totalDaiEarned = swapWethForDai();
+
+        if (totalFxs > 0) {
+            // NOTE: swapping to DAI given treasury decision
+            totalDaiEarned += swapFxsForDai(totalFxs);
+        }
+
+        // Return processed amount
+        processed_ = new TokenAmount[](1);
+        processed_[0] = TokenAmount(address(DAI), totalDaiEarned);
+
+        emit RewardsToStable(address(DAI), totalDaiEarned, block.timestamp);
+    }
+
+    function claimAndRegisterRewards() internal returns (uint256 totalCrv_, uint256 totalCvx_, uint256 totalFxs_) {
+        // Update last claimed time
+        lastClaimTimestamp = block.timestamp;
+
+        uint256 length = baseRewardPools.length();
+        for (uint256 i = 0; i < length;) {
+            IBaseRewardPool baseRewardPool = IBaseRewardPool(baseRewardPools.at(i));
+            if (baseRewardPool.earned(address(this)) > 0) {
+                baseRewardPool.getReward();
+            }
+            unchecked {
+                ++i;
+            }
+        }
+
+        length = pidsPrivateVaults.length();
+        for (uint256 i = 0; i < length;) {
+            address vaultAddr = privateVaults[pidsPrivateVaults.at(i)];
+            IStakingProxy proxy = IStakingProxy(vaultAddr);
+            (, uint256[] memory totalEarned) = proxy.earned();
+            /// NOTE: assume as long at zero index rewards > 0, any other index could be as well
+            if (totalEarned[0] > 0) {
+                proxy.getReward();
+            }
+            unchecked {
+                ++i;
+            }
+        }
+
+        totalCrv_ = CRV.balanceOf(address(this));
+        totalCvx_ = CVX.balanceOf(address(this));
+        totalFxs_ = FXS.balanceOf(address(this));
+
+        if (totalCrv_ == 0) {
+            revert NoRewards();
+        }
+
+        // Emit events for analysis
+        emit RewardClaimed(address(CRV), totalCrv_, block.timestamp);
+        emit RewardClaimed(address(CVX), totalCvx_, block.timestamp);
+        emit RewardClaimed(address(FXS), totalFxs_, block.timestamp);
+    }
+
+    function swapCrvForWeth(uint256 _crvAmount) internal {
+        // Swap CRV -> WETH
+        CRV_ETH_CURVE_POOL.exchange(
+            1, 0, _crvAmount, (getCrvAmountInEth(_crvAmount) * minOutBpsCrvToWeth.val) / MAX_BPS
+        );
+    }
```

Comment:
Let's return the weth amount here and use that instead of doing `WETH.balanceOf`

Reply:
returned and use it then for the other method `weth` for `dai` in 8f1285e8799cd5cb413b8cb597f1829926afab18